main.tf

```diff
@@ -54,18 +55,18 @@ module "rds" {
   backup_retention_period      = var.rds_backup_retention_period
   preferred_backup_window      = var.rds_preferred_backup_window
 
-  subnet_ids         = coalescelist(var.rds_subnet_ids, values(module.network[0].private_subnet_ids))
-  security_group_ids = coalescelist(var.rds_security_group_ids, [module.network[0].database_security_group_id])
+  subnet_ids         = length(var.rds_subnet_ids) > 0 ? var.rds_subnet_ids : values(module.network[0].private_subnet_ids)
+  security_group_ids = length(var.rds_security_group_ids) > 0 ? var.rds_security_group_ids : [module.network[0].database_security_group_id]
```

Comment:
so sad about this, it looked neat :D

Reply:
Yeah - it makes sense though given it's just a function call. Typically function arguments aren't lazily evaluated because you need to know the value to pass through to the function.